packages/@aws-cdk/aws-codedeploy/lambda-packages/ecs-deployment-provider/lib/is-complete.ts

```diff
@@ -0,0 +1,128 @@
+import { Logger } from '@aws-lambda-powertools/logger';
+import { CodeDeploy, DeploymentStatus, GetDeploymentOutput } from '@aws-sdk/client-codedeploy';
+
+const logger = new Logger({ serviceName: 'ecsDeploymentProviderIsComplete' });
+const codedeployClient = new CodeDeploy({});
+
+/**
+ * The request object that the custom resource lamba function receives from CloudFormation.
+ */
+export interface IsCompleteRequest {
+  /**
+   * The type of CloudFormation request (e.g. 'Create', 'Update', or 'Delete')
+   */
+  RequestType: string;
+
+  /**
+   * The physical resource id. Could be undefined for 'Create' events.
```

Comment:
`IsComplete` should be called with the results of the `on-event` right? So the physicalResourceId should never be undefined.

Reply:
Good call. I will update the comments to clarify.